storage/storage_test.go

```diff
@@ -15,3 +15,112 @@ limitations under the License.
 */
 
 package storage_test
+
+import (
+	"testing"
+
+	"github.com/RedHatInsights/insights-results-aggregator/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+func getMockStorage() (storage.Storage, error) {
+	mockStorage, err := storage.New(storage.Configuration{
+		Driver:     "sqlite3",
+		DataSource: ":memory:",
+	})
+	if err != nil {
+		return nil, err
+	}
+	err = mockStorage.Init()
+	if err != nil {
+		return nil, err
+	}
+
+	return mockStorage, nil
+}
+
+func TestMockDBStorageReadReport(t *testing.T) {
+	mockStorage, err := getMockStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mockStorage.Close()
+
+	const testOrgID = storage.OrgID(1)
+	const testClusterName = storage.ClusterName("cluster name")
+	const testClusterReport = storage.ClusterReport("cluster report")
+
+	err = mockStorage.WriteReportForCluster(testOrgID, testClusterName, testClusterReport)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err := mockStorage.ReadReportForCluster(testOrgID, testClusterName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, testClusterReport, result)
+
+}
+
+func TestMockDBStorageListOfOrgs(t *testing.T) {
+	mockStorage, err := getMockStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mockStorage.Close()
+
+	err = mockStorage.WriteReportForCluster(1, "cluster 1", "report 1")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = mockStorage.WriteReportForCluster(3, "cluster 2", "report 2")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err := mockStorage.ListOfOrgs()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, []storage.OrgID{1, 3}, result)
+}
+
+func TestMockDBStorageListOfClustersForOrg(t *testing.T) {
+	mockStorage, err := getMockStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer mockStorage.Close()
+
+	err = mockStorage.WriteReportForCluster(1, "cluster 1", "report 1")
```

Comment:
it works right now, but in the future we need to check if the cluster name is in "GUID"-like format. https://en.wikipedia.org/wiki/Universally_unique_identifier So I'd add that cluster name instead of 'cluster 1'.
Also (it was not mentioned, my fault) - report needs to be in JSON format. It would need to be checked later as well.